Firestore/core/test/firebase/firestore/local/memory_lru_garbage_collector_test.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/local/memory_lru_reference_delegate.h"
+#include "Firestore/core/src/firebase/firestore/local/memory_persistence.h"
+#include "Firestore/core/src/firebase/firestore/model/document_key.h"
+#include "Firestore/core/test/firebase/firestore/local/lru_garbage_collector_test.h"
+#include "Firestore/core/test/firebase/firestore/local/persistence_testing.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace local {
+namespace {
+
+using model::DocumentKey;
+
+class TestHelper : public LruGarbageCollectorTestHelper {
+ public:
+  std::unique_ptr<Persistence> MakePersistence(LruParams lru_params) override {
+    auto persistence = MemoryPersistenceWithLruGcForTesting(lru_params);
+    memory_persistence_ = persistence.get();
+    return std::move(persistence);
+  }
+
+  bool SentinelExists(const DocumentKey& key) override {
+    auto delegate = static_cast<MemoryLruReferenceDelegate*>(
```

Comment:
This sort of assertion is only necessary when capturing a value for later use (e.g. by storing it in a member but not dereferencing it) and it wouldn't otherwise be obvious how the null pointer came about.

In this case, we're not capturing a value and dereferencing the null pointer will crash just about as well as the assertion would, so there's no value in doing this.